src/runtime/GlobalObjectBuiltinString.cpp

```diff
@@ -183,6 +183,25 @@ static Value builtinStringSubstr(ExecutionState& state, Value thisValue, size_t
 
 static Value builtinStringMatch(ExecutionState& state, Value thisValue, size_t argc, Value* argv, bool isNewExpression)
 {
+#ifdef ESCARGOT_ENABLE_ES2015
+    RESOLVE_THIS_BINDING_TO_OBJECT(obj, Object, match);
```

Comment:
This is incorrect.
```
1. Let O be RequireObjectCoercible(this value).
...
4. Let S be ToString(O).
```

So the this value **must** be resolved as string. Also the `String` object has `match` routine, not the `Object.`